### DISCUSSION TITLE
Power BI Service login

### DISCUSSION THREAD

**Discussion by micvar67** [DaxStudio/DaxStudio#588] [Q&A]
Such a cool tool. Thanks for this.

Can you share some info if you have about how login to premium per user power bi works and if it can be called from an app. 

It looks like it calls a windows account login when you shoot of power bi URL - would that be possible to do inside an application ?

**Comment by dgosbell** [comment]
It really depends on what you mean by "app" 

DAX Studio is written in C# using the .Net framework, but as long as you are using some sort of .net language you should be able to do the same thing. I am using the AdomdClient library provided by Microsoft in order to connect. They have versions for both the .net framework (which is windows only) and .net core (which is cross platform)

.Net Framework - https://www.nuget.org/packages/Microsoft.AnalysisServices.AdomdClient.retail.amd64/
.Net Core - https://www.nuget.org/packages/Microsoft.AnalysisServices.AdomdClient.NetCore.retail.amd64

Then it's simply a matter of using a connection like `Data Source=<power bi url>;Initial Catalog=<dataset name>`

The AdomdClient library then handles prompting for credentials if required. However I think that prompting might only work on Windows and on other platforms you would need to do it differently. I did build a little sample console app a little while ago that can run on linux to query a PBI XMLA endpoint.

**Reply by Ljupch0** [reply]
Hi @dgosbell, first off thank you for all the work on DAX Studio. Could it be possible that you publish this linux console app? Querying a PBI XMLA endpoint is exactly what I am trying to do. Thank you!

**Reply by dgosbell** [reply]
While I think it might be technically possible to publish a linux console app since the core ADOMDClient libraries have a .net core version its not something that is easy to do at this point in time. Currently DAX Studio is built using the .net framework which is windows only and the code base is currently not designed for platform independence so there is also some re-architecting that would need to be done to more cleanly separate the query logic from the UI code.

**Reply by Ljupch0** [reply]
Thank you Darren. It does seem that way, even if possible it would be far from straightforward. For now the REST API access should suffice for me.

**Comment by micvar67** [comment]
And if mobile platform ?

When Linux worked you should be able to query XMLA end point on mobile too ?

/m

________________________________
Fra: Darren Gosbell ***@***.***>
Sendt: 9. april 2021 01:10
Til: DaxStudio/DaxStudio ***@***.***>
Cc: Michael Vardinghus ***@***.***>; Author ***@***.***>
Emne: Re: [DaxStudio/DaxStudio] Power BI Service login (#588)


It really depends on what you mean by "app"

DAX Studio is written in C# using the .Net framework, but as long as you are using some sort of .net language you should be able to do the same thing. I am using the AdomdClient library provided by Microsoft in order to connect. They have versions for both the .net framework (which is windows only) and .net core (which is cross platform)

.Net Framework - https://www.nuget.org/packages/Microsoft.AnalysisServices.AdomdClient.retail.amd64/
.Net Core - https://www.nuget.org/packages/Microsoft.AnalysisServices.AdomdClient.NetCore.retail.amd64

Then it's simply a matter of using a connection like Data Source=<power bi url>;Initial Catalog=<dataset name>

The AdomdClient library then handles prompting for credentials if required. However I think that prompting might only work on Windows and on other platforms you would need to do it differently. I did build a little sample console app a little while ago that can run on linux to query a PBI XMLA endpoint.

**Reply by dgosbell** [reply]
I don't really know, I have not done any serious work with mobile apps. If you are building using .net it might be possible, although you would probably need to manually build the authentication prompts to get the OAuth tokens to connect to PBI. 

Microsoft have a Power BI mobile app that can display Power BI reports so it should be technically possible.
